api/src/API/Controllers/RaceResultsController.cs

```diff
@@ -51,12 +51,29 @@ public async Task<IActionResult> GetAllRaceResults(string orgId)
             var racesNeeded = raceResults.Select(result => result.RaceId).ToHashSet();
             var racesInResponse = await raceContainer.GetManyAsDictAsync(it => it.Where(race => racesNeeded.Contains(race.Id)));
 
-            var result = raceResults.Select(raceResult => new RaceResultResponse()
+            var result = new List<RaceResultResponse>();
+            foreach (var raceResult in raceResults)
             {
-                RaceResult = raceResult,
-                Member = members[raceResult.MemberId],
-                Race = racesInResponse[raceResult.RaceId],
-            });
+                Member? memberForRaceResult = null;
```

Comment:
This alternative approach still results in a silent failure. There's never an exception thrown even though we should expect this to never happen (because of the invariants mentioned in the previous discussion)

Reply:
This is not a silent failure - the client will see that this data is missing and present it to the user. We can add some kind of AI logging here if you'd like, too. But we should not bring down the route if there's a problem here (see my comment above about why - there are countless scenarios besides "we messed up deleting something" where this could be happening).

Reply:
I agree we shouldn't block the route. Let's just check for null on the client then